core/ir/aarch64/instr.c

```diff
@@ -737,3 +739,121 @@ instr_invert_predicate(dr_pred_type_t pred)
     default: CLIENT_ASSERT(false, "Incorrect predicate value"); return DR_PRED_NONE;
     }
 }
+
+ptr_int_t
+compute_scaled_index_aarch64(opnd_t opnd, reg_t index_val)
+{
+    bool scaled = false;
+    uint amount = 0;
+    dr_extend_type_t type = opnd_get_index_extend(opnd, &scaled, &amount);
+    reg_t extended = 0;
+    uint msb = 0;
+    switch (type) {
+    default: CLIENT_ASSERT(false, "Unsupported extend type"); return 0;
+    case DR_EXTEND_UXTW: extended = (index_val << (63u - 31u)) >> (63u - 31u); break;
```

Comment:
I didn't write this code; I just moved it from `opnd_shared.c` so it can be shared with `compute_scaled_index_aarch64()`. This was commented on in the PR that originally introduced it:
https://github.com/DynamoRIO/dynamorio/pull/4529#discussion_r537517400

I have removed it now anyway and changed it to use a mask as per your other comment.